File: src/main/java/message/Messages.java
/*
 * Copyright (c) Grigory "vcusnx" Markovich, 2023.
 *
 * The Messenger Project.
 */

package message;

import user.User;


public class Messages implements MessagesManager {

    private final long messageId;
    private final long chatId;
    private final String userName;
    private final String messageText;

    /**
     * Constructor for Message class
     */
    public Messages() {
        this.chatId = getChatId();
        this.messageId = getMessageId();
        this.userName = getUserName();
        this.messageText = getMessageText();
    }

    /**
     * @return Chat ID
     */
    public long getChatId() {
        return chatId;
    }

    /**
     * @return Message ID
     */
    public long getMessageId() {
        return messageId;
    }

    /**
     * @return User name
     */
    public String getUserName() {
        return userName;
    }

    /**
     * @return Message text
     */
    public String getMessageText() {
        return messageText;
    }

    @Override
    public void send(long messageId, User user, String text) {

    }

    @Override
    public void delete(long messageId, User user) {

    }

    @Override
    public void edit(long messageId, User user, String text) {

    }
}

File: app/src/main/kotlin/message/Messages.kt
/*
 * Copyright (c) Grigory "vcusnx" Markovich, ISLANDS TEAM, 2023.
 *
 *
 */

package message

import user.User

class Messages: MessagesManager {
    /**
     * Sends message
     *
     * @param messageId message ID
     * @param user message from User
     * @param text message text
     */
    override fun send(messageId: Long, user: String, text: String) {
        TODO("Not yet implemented")
    }

    /**
     * Deletes message
     *
     * @param messageId message ID
     * @param user message from User
     */
    override fun delete(messageId: Long, user: User) {
        TODO("Not yet implemented")
    }

    /**
     * Edits message
     *
     * @param messageId message ID
     * @param user message from User
     * @param text new message text
     */
    override fun edit(messageId: Long, user: User, text: String) {
        TODO("Not yet implemented")
    }

}